File: 3.0.x - 3.4.x/Yodo1Ads.java
//
// Created by Yodo1 on 21/07/2022.
//

package com.cocos.game;

import android.content.Context;
import android.os.Debug;
import android.view.Gravity;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.RelativeLayout;

import androidx.annotation.NonNull;

import com.cocos.lib.CocosHelper;
import com.yodo1.mas.Yodo1Mas;
import com.yodo1.mas.banner.Yodo1MasBannerAdListener;
import com.yodo1.mas.banner.Yodo1MasBannerAdSize;
import com.yodo1.mas.banner.Yodo1MasBannerAdView;
import com.yodo1.mas.error.Yodo1MasError;
import com.yodo1.mas.helper.model.Yodo1MasAdBuildConfig;
import com.yodo1.mas.interstitial.Yodo1MasInterstitialAd;
import com.yodo1.mas.interstitial.Yodo1MasInterstitialAdListener;
import com.yodo1.mas.reward.Yodo1MasRewardAd;
import com.yodo1.mas.reward.Yodo1MasRewardAdListener;
import com.cocos.lib.CocosJavascriptJavaBridge;
public class Yodo1Ads {

    private static Yodo1MasBannerAdView bannerAdView;
    private static AppActivity activity;

    public void initSDK(AppActivity activity) {
        this.activity = activity;
    }
    public static void initializeSdk(AppActivity activity, String appKey,boolean isEnabled) //its working
    {

        Yodo1Ads yodo1Ads = new Yodo1Ads();
        yodo1Ads.initSDK(activity);

        activity.runOnUiThread( new Runnable() {
            public void run() {
                if(isEnabled) {
                    Yodo1MasAdBuildConfig config = new Yodo1MasAdBuildConfig.Builder().enableUserPrivacyDialog(true).build();
                    Yodo1Mas.getInstance().setAdBuildConfig(config);
                }
                Yodo1Mas.getInstance().initMas(activity, appKey, new Yodo1Mas.InitListener() {
                    @Override
                    public void onMasInitSuccessful() {
                       // nativeOnMasInitSuccessful();
                    }

                    @Override
                    public void onMasInitFailed(@NonNull Yodo1MasError error) {
                        //nativeOnMasInitFailed();
                    }
                });
            }
         });
    }

    public static void initializeInterstitialAds()
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1MasInterstitialAd.getInstance().loadAd(activity);

                Yodo1MasInterstitialAd.getInstance().setAdListener(new Yodo1MasInterstitialAdListener() {

                    @Override
                    public void onInterstitialAdLoaded(Yodo1MasInterstitialAd ad) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("find(\"Yodo1\").getComponent(\"Yodo1Ads\").onInterstitialAdLoaded()");
                            }
                        });
                    }

                    @Override
                    public void onInterstitialAdFailedToLoad(Yodo1MasInterstitialAd ad, @NonNull Yodo1MasError error) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("find(\"Yodo1\").getComponent(\"Yodo1Ads\").onInterstitialAdFailedToLoad()");
                            }
                        });
                    }

                    @Override
                    public void onInterstitialAdOpened(Yodo1MasInterstitialAd ad) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("find(\"Yodo1\").getComponent(\"Yodo1Ads\").onInterstitialAdOpened()");
                            }
                        });
                    }

                    @Override
                    public void onInterstitialAdFailedToOpen(Yodo1MasInterstitialAd ad, @NonNull Yodo1MasError error) {
                        ad.loadAd(activity);
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("find(\"Yodo1\").getComponent(\"Yodo1Ads\").onInterstitialAdFailedToOpen()");
                            }
                        });
                    }

                    @Override
                    public void onInterstitialAdClosed(Yodo1MasInterstitialAd ad) {
                        ad.loadAd(activity);
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("find(\"Yodo1\").getComponent(\"Yodo1Ads\").onInterstitialAdClosed()");
                            }
                        });
                    }
                });
            }
        });

    }

    public static void showInterstitialAds()
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1MasInterstitialAd.getInstance().showAd(activity);
            }
        });
    }

    public static void initializeRewardAds() {

        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1MasRewardAd.getInstance().loadAd(activity);

                Yodo1MasRewardAd.getInstance().setAdListener(new Yodo1MasRewardAdListener() {

                    @Override
                    public void onRewardAdLoaded(Yodo1MasRewardAd ad) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onRewardAdLoaded()");
                            }
                        });
                    }

                    @Override
                    public void onRewardAdFailedToLoad(Yodo1MasRewardAd ad, @NonNull Yodo1MasError error) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onRewardAdFailedToLoad()");
                            }
                        });
                    }

                    @Override
                    public void onRewardAdOpened(Yodo1MasRewardAd ad) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onRewardAdOpened()");
                            }
                        });

                    }

                    @Override
                    public void onRewardAdFailedToOpen(Yodo1MasRewardAd ad, @NonNull Yodo1MasError error) {
                        ad.loadAd(activity);
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onRewardAdFailedToOpen()");
                            }
                        });
                    }

                    @Override
                    public void onRewardAdClosed(Yodo1MasRewardAd ad) {
                        ad.loadAd(activity);
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onRewardAdClosed()");
                            }
                        });
                    }

                    @Override
                    public void onRewardAdEarned(Yodo1MasRewardAd ad) {

                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onRewardAdEarned()");
                            }
                        });

                    }
                });
            }
        });


    }

    public static void showRewardAds()
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1MasRewardAd.getInstance().showAd(activity);
            }
        });
    }



    public static void loadBannerAds(String size, String horizontal_alignment, String vertical_alignment)
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {

                bannerAdView = new Yodo1MasBannerAdView(activity);

                if(size.equalsIgnoreCase( "Banner"))
                    bannerAdView.setAdSize(Yodo1MasBannerAdSize.Banner);
                else if(size.equalsIgnoreCase( "AdaptiveBanner"))
                    bannerAdView.setAdSize(Yodo1MasBannerAdSize.AdaptiveBanner);
                else if(size.equalsIgnoreCase( "LargeBanner"))
                    bannerAdView.setAdSize(Yodo1MasBannerAdSize.LargeBanner);
                else if(size.equalsIgnoreCase( "SmartBanner"))
                    bannerAdView.setAdSize(Yodo1MasBannerAdSize.SmartBanner);
                else if(size.equalsIgnoreCase( "IABMediumRectangle"))
                    bannerAdView.setAdSize(Yodo1MasBannerAdSize.IABMediumRectangle);

                RelativeLayout adRelativeLayout = new RelativeLayout(activity);

                if(horizontal_alignment.equalsIgnoreCase("CENTER"))
                    adRelativeLayout.setHorizontalGravity(Gravity.CENTER_HORIZONTAL);
                else if(horizontal_alignment.equalsIgnoreCase("left"))
                    adRelativeLayout.setHorizontalGravity(Gravity.LEFT);
                else if(horizontal_alignment.equalsIgnoreCase("right"))
                    adRelativeLayout.setHorizontalGravity(Gravity.RIGHT);

                if(vertical_alignment.equalsIgnoreCase("TOP"))
                    adRelativeLayout.setVerticalGravity(Gravity.TOP);
                else if(vertical_alignment.equalsIgnoreCase("BOTTOM"))
                    adRelativeLayout.setVerticalGravity(Gravity.BOTTOM);

                adRelativeLayout.addView(bannerAdView);

                bannerAdView.setAdListener(new Yodo1MasBannerAdListener() {
                    @Override public void onBannerAdLoaded(Yodo1MasBannerAdView bannerAdView) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onBannerAdLoaded()");
                            }
                        });
                    }
                    @Override
                    public void onBannerAdFailedToLoad(Yodo1MasBannerAdView bannerAdView, @NonNull Yodo1MasError error) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onBannerAdFailedToLoad()");
                            }
                        });
                    }
                    @Override public void onBannerAdOpened(Yodo1MasBannerAdView bannerAdView) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onBannerAdOpened()");
                            }
                        });
                    }
                    @Override
                    public void onBannerAdFailedToOpen(Yodo1MasBannerAdView bannerAdView, @NonNull Yodo1MasError error) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onBannerAdFailedToOpen()");
                            }
                        });
                    }
                    @Override
                    public void onBannerAdClosed(Yodo1MasBannerAdView bannerAdView) {
                        CocosHelper.runOnGameThread(new Runnable() {
                            public void run() {
                                CocosJavascriptJavaBridge.evalString("cc.find(\"Yodo1\").getComponent(\"Yodo1Ads\").onBannerAdClosed()");
                            }
                        });
                    }
                });
                FrameLayout rootView = activity.findViewById(android.R.id.content);

                rootView.addView(adRelativeLayout);
                //activity.setContentView(adRelativeLayout);
                bannerAdView.loadAd();
            }
        });
    }

    public static void hideBannerAds() {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                if (bannerAdView.getVisibility() == View.VISIBLE) {
                    bannerAdView.setVisibility(View.GONE);
                }
            }
        });
    }

    public static void showBannerAds() {

        activity.runOnUiThread( new Runnable() {
            public void run() {
                if (bannerAdView.getVisibility() == View.GONE) {
                    bannerAdView.setVisibility(View.VISIBLE);
                }
            }
        });
    }

    public static void setCOPPA(boolean isEnabled)
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1Mas.getInstance().setCOPPA(isEnabled);
            }
        });
    }

    public static void setGDPR(boolean isEnabled)
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1Mas.getInstance().setGDPR(isEnabled);
            }
        });
    }

    public static void setCCPA(boolean isEnabled)
    {
        activity.runOnUiThread( new Runnable() {
            public void run() {
                Yodo1Mas.getInstance().setCCPA(isEnabled);
            }
        });
    }



}

File: 3.0.x - 3.4.x/Yodo1Ads.ts
// Learn TypeScript:
//  - https://docs.cocos.com/creator/2.4/manual/en/scripting/typescript.html
// Learn Attribute:
//  - https://docs.cocos.com/creator/2.4/manual/en/scripting/reference/attributes.html
// Learn life-cycle callbacks:
//  - https://docs.cocos.com/creator/2.4/manual/en/scripting/life-cycle-callbacks.html

import { _decorator, Component, Node, debug, DebugMode, log, find, System, sys } from 'cc';




const { ccclass, property } = _decorator;



@ccclass
export default class Yodo1Ads extends Component {

   

   // @property
    //text: string = 'hello';
    
   
    // LIFE-CYCLE CALLBACKS:

    private static _instance:Yodo1Ads = null;

    public static getInstance(){
        if(this._instance==null) this._instance = new Yodo1Ads();
        return this._instance;
    }

    public setCOPPA (isEnabled)
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "setCOPPA:", isEnabled);
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "setCOPPA", "(Z)V",isEnabled);
    }

    public setGDPR (isEnabled)
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "setGDPR:", isEnabled);
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "setGDPR", "(Z)V",isEnabled);
    }

    public setCCPA (isEnabled)
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "setCCPA:", isEnabled);
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "setCCPA", "(Z)V",isEnabled);
    }


    public initializeMasSdk (appkey,isEnabled)
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "initializeSdkWithAppKey:isEnabled:", appkey, isEnabled);

        }
        else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "initializeSdk", "(Ljava/lang/String;Z)V",appkey,isEnabled);
    }

    public initializeInterstitialAds() 
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "initializeInterstitialAds");
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "initializeInterstitialAds", '()V');

    }

    public showInterstitialAds() 
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "showInterstitialAds");
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "showInterstitialAds", '()V');

    }

    public initializeRewardAds() 
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "initializeRewardAds");
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "initializeRewardAds", '()V');

    }

    public showRewardAds() 
    {
        if(sys.isNative && (sys.os == sys.OS.IOS || sys.os == sys.OS.OSX)){
            jsb.reflection.callStaticMethod("Yodo1Ads", "showRewardAds");
       }
       else
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "showRewardAds", '()V');

    }

    public loadBannerAds(size, horizontalAlignment, verticalAlignment) 
    {
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "loadBannerAds", '(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V',size,horizontalAlignment,verticalAlignment);

    }

    public hideBannerAds() 
    {
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "hideBannerAds", '()V');

    }

    public showBannerAds() 
    {
        jsb.reflection.callStaticMethod("com/cocos/game/AppActivity", "showBannerAds", '()V');

    }

    //Interstitial Ad events 

    public onInterstitialAdLoaded() {

    }

    public onInterstitialAdFailedToLoad() {
        
    }

    public onInterstitialAdOpened() {
       
    }

    
    public onInterstitialAdFailedToOpen() {
       
        
    }
   
    public onInterstitialAdClosed() {
        
        
    }

    // Rewarded Ad events. 

    public onRewardAdLoaded() {
       
        
    }

    
    public onRewardAdFailedToLoad() {
        
    }

   
    public  onRewardAdOpened() {
        

    }

    public onRewardAdFailedToOpen() {
      
    }

    public  onRewardAdClosed() {
       
    }

    public onRewardAdEarned() {
       // Reward user here
     

    }

    //Banner add events
    public onBannerAdLoaded() {

    }

    public onBannerAdFailedToLoad() {

    }

    public onBannerAdOpened() {

    }

    public onBannerAdFailedToOpen() {

    }

    public onBannerAdClosed() {

    }


   

}

(window as any).Yodo1Ads = Yodo1Ads;



